d2bs/kolbot/libs/common/Attack.js

```diff
@@ -779,12 +781,17 @@ var Attack = {
 
 			if (result) {
 				Pather.moveTo(result[0], result[1], 3, spectype);
+				previous = result;
 				//this.countUniques();
 
 				if (!this.clear(40, spectype)) {
 					break;
 				}
 			}
+			// Make sure bot does not get stuck in different area.
+			else if (currentArea !== getArea().id) {
```

Comment:
Should this be else if or just if?
For example, if this.clear is successful, but results in player leaving the area (chasing mobs), we would still want it to return to previous location right?

Reply:
This should be an else if. Current area only needs to be checked when result is false. In the example when this.clear is successful the player may be stuck in another area. The next iteration of the while loop will calculate result as false then move the player back to the previous room.